src/solver/type_getters.jl

```diff
@@ -477,6 +482,54 @@ thermo_state_type(::EquilMoistModel, ::Type{FT}) where {FT} = TD.PhaseEquil{FT}
 thermo_state_type(::NonEquilMoistModel, ::Type{FT}) where {FT} =
     TD.PhaseNonEquil{FT}
 
+function get_restart_file(
+    output_dir_style::OutputPathGenerator.OutputPathGeneratorStyle,
+    base_output_dir;
+    restart_file = nothing,
+    detect_restart_file = false,
+)
+    detect_restart_file &&
+        error("detect_restart_file works only with ActiveLink")
+    return restart_file
+end
+
+function get_restart_file(
+    output_dir_style::OutputPathGenerator.ActiveLinkStyle,
+    base_output_dir;
+    restart_file = nothing,
+    detect_restart_file = false,
+)
```

Comment:
I like this version! One minor thing is that `@assert isdir(base_output_dir)` is not correct: it is fine to not have a `base_output_dir` if this is the first time we run this simulation (it will be created). In that case, `restart_file` should be nothing.

Reply:
This is the updated version:
```julia
function auto_detect_restart_file(
    output_dir_style::OutputPathGenerator.ActiveLinkStyle,
    base_output_dir
)
    # if base_output_dir does not exist, we return restart_file = nothing because there is
    # no restart file to be detected
    isdir(base_output_dir) || return nothing

    # output_dir will be something like ABC/DEF/output_1234
    name_rx = r"output_(\d\d\d\d)"
    restart_file_rx = r"day\d+\.\w+\.hdf5"
    restart_file = nothing

    existing_outputs =
        filter(x -> !isnothing(match(name_rx, x)), readdir(base_output_dir))

    isempty(existing_outputs) && return nothing

    latest_output = first(sort(existing_outputs, rev = true))
    previous_folder = joinpath(base_output_dir, latest_output)
    possible_restart_files = filter(
        f -> occursin(restart_file_rx, f),
        readdir(previous_folder),
    )
    if isempty(possible_restart_files)
        @warn "Detected folder $(previous_folder), but no restart file was found"
        return nothing
    end

    restart_file_name =
        last(CA.sort_files_by_time(possible_restart_files))
    restart_file = joinpath(previous_folder, restart_file_name)
    @assert isfile(restart_file) "Restart file does not exist"

    return restart_file
end
```

I also removed the other assert because it is also fine to start a new simulation if no restart files were found